app/javascript/lib/Entities/AbstractImperialGame.js

```diff
@@ -8,14 +8,42 @@ export default class AbstractImperialGame extends Entity {
   }
 
   #rondel;
+  #availableFreeRondelSlotCount = 3;
+  #availablePaidRondelSlotCount = 3;
 
-  constructor(id) {
+  #nations;
+  #nationOrder;
+
+  constructor(id, nations, nationOrder) {
     super(id);
 
     this.#rondel = new Rondel();
+
+    this.#nations = nations;
+    this.#nationOrder = nationOrder;
   }
 
   get rondel() {
     return this.#rondel;
   }
+
+  get availableFreeRondelSlotCount() {
+    return this.#availableFreeRondelSlotCount;
+  }
+  get availablePaidRondelSlotCount() {
+    return this.#availablePaidRondelSlotCount;
+  }
+
+  nationIdToEntity(nationId) {
+    return this.#nations.get(nationId);
+  }
+
+  nationTurnAfter(nation) {
```

Comment:
Next minor refactor I'll probably start moving out `nextNation` logic from `imperial.js`
After that will probably be the `Taxation` rondel slot

We'll probably start seeing major reductions to `imperial.js` once those happen